### PR TITLE
refactor ptq exporter pytorch with activation holder

### PR DESCRIPTION
**Pull request by eladc-git**

### CODE REVIEW

File: model_compression_toolkit/exporter/model_wrapper/pytorch/builder/fully_quantized_model_builder.py [hunk 1]
```diff
@@ -37,10 +40,29 @@ def fully_quantized_wrapper(node: common.BaseNode,
...
         """
-        weight_quantizers, activation_quantizers = get_quantization_quantizers(node)
-        wrapped_layer = PytorchQuantizationWrapper(module, weight_quantizers, activation_quantizers)
-        return wrapped_layer
+        weight_quantizers, _ = get_quantization_quantizers(node)
+        if len(weight_quantizers) > 0:
+            return PytorchQuantizationWrapper(module, weight_quantizers)
+        return module
 
+    def get_activation_quantizer_holder(node: BaseNode) -> Union[None, Callable]:
```
Comment (reuvenperetz) [inline model_compression_toolkit/exporter/model_wrapper/pytorch/builder/fully_quantized_model_builder.py:48]: Fix type hints

File: tests/pytorch_tests/function_tests/test_fully_quantized_exporter.py [hunk 2]
```diff
@@ -100,41 +100,28 @@ def test_layers_wrapper(self):
         self.assertTrue(self.fully_quantized_mbv2.features_0_0_bn.is_weights_quantization)
         self.assertFalse(self.fully_quantized_mbv2.features_0_0_bn.is_activation_quantization)
 
-        self.assertTrue(isinstance(self.fully_quantized_mbv2.features_0_2, PytorchQuantizationWrapper))
-        self.assertTrue(isinstance(self.fully_quantized_mbv2.features_0_2.layer, torch.nn.ReLU6))
-        self.assertFalse(self.fully_quantized_mbv2.features_0_2.is_weights_quantization)
-        self.assertTrue(self.fully_quantized_mbv2.features_0_2.is_activation_quantization)
```
Comment (reuvenperetz) [inline tests/pytorch_tests/function_tests/test_fully_quantized_exporter.py:106]: Check that relu6 activations are quantized

File: tests/pytorch_tests/function_tests/test_fully_quantized_exporter.py [hunk 3]
```diff
@@ -100,41 +100,28 @@ def test_layers_wrapper(self):
...
-
-        self.assertTrue(isinstance(self.fully_quantized_mbv2.classifier_0, PytorchQuantizationWrapper))
-        self.assertTrue(isinstance(self.fully_quantized_mbv2.classifier_0.layer, torch.nn.Dropout))
-        self.assertFalse(self.fully_quantized_mbv2.classifier_0.is_weights_quantization)
-        self.assertFalse(self.fully_quantized_mbv2.classifier_0.is_activation_quantization)
```
Comment (reuvenperetz) [inline tests/pytorch_tests/function_tests/test_fully_quantized_exporter.py:111]: Check that dropout is not quantized

File: tests/pytorch_tests/function_tests/test_fully_quantized_exporter.py [hunk 4]
```diff
@@ -100,41 +100,28 @@ def test_layers_wrapper(self):
...
-        self.assertTrue(len(self.fully_quantized_mbv2.features_0_2.activation_quantizers) == 1)
-        self.assertTrue(isinstance(self.fully_quantized_mbv2.features_0_2.activation_quantizers[0],
+        self.assertTrue(isinstance(self.fully_quantized_mbv2.features_0_2_activation_holder_quantizer.activation_holder_quantizer,
                                    ActivationPOTInferableQuantizer))
 
     def test_no_quantization_qc(self):
-        self.assertTrue(len(self.fully_quantized_mbv2.classifier_0.activation_quantizers) == 0)
-        self.assertTrue(len(self.fully_quantized_mbv2.classifier_0.weights_quantizers) == 0)
+        self.assertTrue(isinstance(self.fully_quantized_mbv2.features_0_2, torch.nn.ReLU6))
+        self.assertTrue(isinstance(self.fully_quantized_mbv2.classifier_0, torch.nn.Dropout))
```
Comment (reuvenperetz) [inline tests/pytorch_tests/function_tests/test_fully_quantized_exporter.py:124]: These checks does not assert these layers are not quantized

**Review by reuvenperetz — CHANGES_REQUESTED** [review]

**Review by ofirgo — APPROVED** [review]

File: model_compression_toolkit/exporter/model_wrapper/pytorch/builder/fully_quantized_model_builder.py (hunk 1 above)
Comment (eladc-git) [reply model_compression_toolkit/exporter/model_wrapper/pytorch/builder/fully_quantized_model_builder.py:48]: fix to what? it is Union[None, Callable]

File: tests/pytorch_tests/function_tests/test_fully_quantized_exporter.py (hunk 2 above)
Comment (eladc-git) [reply tests/pytorch_tests/function_tests/test_fully_quantized_exporter.py:106]: already did in this test "test_activation_qc"

File: tests/pytorch_tests/function_tests/test_fully_quantized_exporter.py (hunk 4 above)
Comment (eladc-git) [reply tests/pytorch_tests/function_tests/test_fully_quantized_exporter.py:124]: so the assert to check these layer are not holders or wrappers just simple layers

File: tests/pytorch_tests/function_tests/test_fully_quantized_exporter.py (hunk 3 above)
Comment (eladc-git) [reply tests/pytorch_tests/function_tests/test_fully_quantized_exporter.py:111]: how do you do that? the only thing I thought is to check that this member "classifier_0_activation_holder_quantizer" does not exist.

**Review by reuvenperetz — APPROVED** [review]
